nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/impl/ACPImpl.java

```diff
@@ -86,6 +87,9 @@ public void addACL(ACL acl) {
                     ACL local = getACL(ACL.LOCAL_ACL);
                     if (local != null) {
                         int i = acls.indexOf(local);
+                        if (Framework.isBooleanPropertyTrue(LEGACY_BEHAVIOUR_PROPERTY)) {
+                            ++i;
```

Comment:
You could use `i++` instead of `++i` here...
What if the local ACL is the last one and so `i++` is out of bound?

Reply:
`i++` is the standard Java idiom, please use it (we're not counting cycles and this is not C and anyway I doubt your assertion, the JIT will optimize both in the same way).